cpp/src/rolling/range_window_bounds.cpp

```diff
@@ -30,36 +30,28 @@ namespace {
  * This makes it possible to copy construct and copy assign `range_window_bounds` objects.
  */
 struct range_scalar_constructor {
-  template <typename T, CUDF_ENABLE_IF(not detail::is_supported_range_type<T>())>
+  template <typename T>
   std::unique_ptr<scalar> operator()(scalar const& range_scalar_) const
   {
-    CUDF_FAIL(
-      "Unsupported range type. "
-      "Only Durations, fixed-point, and non-boolean integral range types are allowed.");
-  }
-
-  template <typename T, CUDF_ENABLE_IF(cudf::is_duration<T>())>
-  std::unique_ptr<scalar> operator()(scalar const& range_scalar_) const
-  {
-    return std::make_unique<duration_scalar<T>>(
-      static_cast<duration_scalar<T> const&>(range_scalar_));
-  }
-
-  template <typename T, CUDF_ENABLE_IF(std::is_integral_v<T> && not cudf::is_boolean<T>())>
-  std::unique_ptr<scalar> operator()(scalar const& range_scalar_) const
-  {
-    return std::make_unique<numeric_scalar<T>>(
-      static_cast<numeric_scalar<T> const&>(range_scalar_));
-  }
-
-  template <typename T, CUDF_ENABLE_IF(cudf::is_fixed_point<T>())>
-  std::unique_ptr<scalar> operator()(scalar const& range_scalar_) const
-  {
-    return std::make_unique<fixed_point_scalar<T>>(
-      static_cast<fixed_point_scalar<T> const&>(range_scalar_));
+    if constexpr (std::is_floating_point_v<T> ||
+                  (std::is_integral_v<T> && not cudf::is_boolean<T>())) {
+      return std::make_unique<numeric_scalar<T>>(
+        static_cast<numeric_scalar<T> const&>(range_scalar_));
+    } else if constexpr (cudf::is_fixed_point<T>()) {
+      return std::make_unique<fixed_point_scalar<T>>(
+        static_cast<fixed_point_scalar<T> const&>(range_scalar_));
+    } else if constexpr (cudf::is_duration<T>()) {
+      return std::make_unique<duration_scalar<T>>(
+        static_cast<duration_scalar<T> const&>(range_scalar_));
+    } else {
+      assert(not detail::is_supported_range_type<T>());
+      CUDF_FAIL(
+        "Unsupported range type. "
+        "Only Durations, fixed-point, floating point, and non-boolean integral range types are "
+        "allowed.");
+    }
```

Comment:
I prefer the original specialization approach over this if-else approach. Is there some reason that just adding
```
  template <typename T, CUDF_ENABLE_IF(cudf::is_floating_point<T>())>
  std::unique_ptr<scalar> operator()(scalar const& range_scalar_) const
  {
    return std::make_unique<numeric_scalar<T>>(
      static_cast<numeric_scalar<T> const&>(range_scalar_));
  }
```
would not work here?

Reply:
I agree that this chain of `if/else` looks worse than the previous approach :smiley:

Reply:
I've changed it back now.